alibuild_helpers/build.py

```diff
@@ -344,33 +344,30 @@ def doBuild(args, parser):
                   "  git pull --rebase\n",
                   pwd=os.getcwd(), star=star()))
 
-  # Prefetch git heads in parallel
-  debug("Prefetching git heads in parallel:")
-
-  fetchers = list(filter(lambda p: "source" in specs[p], buildOrder))
-  poolConfig = {"processes": min(max(len(fetchers), 1), 16)}
-  pool = ThreadPool(**poolConfig)
-  debug("Created thread pool with config: %s" % poolConfig)
-
-  git_commands = {} # store futures
-  for p in fetchers:
-    # Replace source with local one if we are in development mode.
+  # Clone/update repos
+  reposWithSource = list(filter(lambda p: "source" in specs[p], buildOrder))
+  if args.offline:
+    banner("Offline mode - not attempting to update any repositories\n")
+    for p in reposWithSource:
+      specs[p]["reference"] = "%s/%s" % (abspath(args.referenceSources), p.lower())
+      dieOnError(not exists(specs[p]["reference"]), "Local clone of %s does not exist yet, run again without --offline option.");
+  else:
+    banner("Cloning/Updating repositories\nUse --offline to skip this step. Skipping will only work after the first\nsuccessful completion of this step without --offline.\n")
```

Comment:
Do we really need this printed every time as a banner?

Reply:
My thinking of making it a banner was, that it is a step that takes a long time. At least in the inverted case (offline mode enabled), @sawenzel suggested to make it very clear in the output.

I leave the decision to you, I don't mind printing in a different severity level.

Reply:
Please leave it out for now. We should leave the old behavior unchanged for the moment.

Reply:
Done